rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -3531,18 +3539,100 @@ void main_window::SetIconSizeActions(int idx) const
 		ui->setIconSizeLargeAct->setChecked(true);
 }
 
-void main_window::RemoveDiskCache()
+void main_window::RemoveHDD1Caches()
+{
+	if (fs::remove_all(rpcs3::utils::get_hdd1_dir() + "caches", false))
+	{
+		QMessageBox::information(this, tr("HDD1 Caches Removed"), tr("HDD1 caches successfully removed"));
+	}
+	else
+	{
+		QMessageBox::warning(this, tr("Error"), tr("Could not remove HDD1 caches"));
+	}
+}
+
+void main_window::RemoveAllCaches()
+{
+	if (QMessageBox::question(this, tr("Confirm Removal"), tr("Remove all caches?")) != QMessageBox::Yes)
+		return;
+
+	const std::string cache_base_dir = rpcs3::utils::get_cache_dir();
+	u64 caches_count = 0;
+	u64 caches_removed = 0;
+
+	for (const game_info& game : m_game_list_frame->GetGameInfo()) // Loop on discovered games
+	{
+		if (game && qstr(game->info.category) != cat::cat_ps3_os && fs::exists(cache_base_dir + game->info.serial)) // If not OS category and cache exists
+		{
+			caches_count++;
+
+			if (fs::remove_all(cache_base_dir + game->info.serial))
+			{
+				caches_removed++;
+			}
+		}
+	}
+
+	if (caches_count == caches_removed)
+	{
+		QMessageBox::information(this, tr("Caches Removed"), tr("%0 cache(s) successfully removed").arg(caches_removed));
+	}
+	else
+	{
+		QMessageBox::warning(this, tr("Error"), tr("Could not remove %0 of %1 cache(s)").arg(caches_count - caches_removed).arg(caches_count));
+	}
+
+	RemoveHDD1Caches();
+}
+
+void main_window::RemoveSavestates()
 {
-	const std::string cache_dir = rpcs3::utils::get_hdd1_dir() + "/caches";
+	if (QMessageBox::question(this, tr("Confirm Removal"), tr("Remove savestates?")) != QMessageBox::Yes)
+		return;
 
-	if (fs::remove_all(cache_dir, false))
+	if (fs::remove_all(fs::get_config_dir() + "savestates", false))
 	{
-		QMessageBox::information(this, tr("Cache Cleared"), tr("Disk cache was cleared successfully"));
+		QMessageBox::information(this, tr("Savestates Removed"), tr("Savestates successfully removed"));
 	}
 	else
 	{
-		QMessageBox::warning(this, tr("Error"), tr("Could not remove disk cache"));
+		QMessageBox::warning(this, tr("Error"), tr("Could not remove savestates"));
+	}
+}
+
+void main_window::CleanupGamesList()
+{
+	if (QMessageBox::question(this, tr("Confirm Removal"), tr("Remove invalid game paths from games list?\n" \
+		"Undetectable (zombies) and/or corrupted games will be removed from games list \"games.yml\" file")) != QMessageBox::Yes)
```

Comment:
I guess more like this
```suggestion
		"Undetectable games (zombies) as well as corrupted games will be removed from the game list file (\"games.yml\").")) != QMessageBox::Yes)
```